packages/build-utils/src/workspaces/get-workspaces.ts

```diff
@@ -0,0 +1,65 @@
+import { DetectorFilesystem } from '../detectors/filesystem';
+import { workspaceManagers } from './workspace-managers';
+import { detectFramework as detectWorkspaceManagers } from '../detect-framework';
+
+const MAX_DEPTH_TRAVERSE = 2;
+
+export interface GetWorkspaceOptions {
+  fs: DetectorFilesystem;
+}
+
+export type WorkspaceType = 'yarn' | 'pnpm' | 'npm';
+
+export type Workspace = {
+  implementation: WorkspaceType;
+  rootPath: string;
+};
+
+export async function getWorkspaces({
+  fs,
+}: GetWorkspaceOptions): Promise<Workspace[]> {
+  const rootWorkspaceImplementation = await detectWorkspaceManagers({
+    fs,
+    frameworkList: workspaceManagers,
+  });
+
+  if (rootWorkspaceImplementation === null) {
+    const workspaces: Array<Workspace> = [];
+    const directoryContents = await fs.readdir('./');
+    const childDirectories = directoryContents.filter(
+      stat => stat.type === 'dir'
+    );
+    let currentDepth = 1;
+
+    while (currentDepth++ < MAX_DEPTH_TRAVERSE) {
```

Comment:
```suggestion
    for (let currentDepth = 0; currentDepth < MAX_DEPTH_TRAVERSE; currentDepth++) {
```

Let's use a traditional for loop?